### ISSUE TITLE
Portal doesn't correctly process homologues

### ISSUE THREAD

**Issue by julie-sullivan** [intermine/intermine#976]
This line doesn't work as expected:

https://github.com/intermine/intermine/blob/beta/bio/webapp/resources/webapp/model/js/friendly-mines.js#L104

It's not keeping track of the domains properly.

e.g. In HumanMine, sending MGI identifiers to MouseMine includes the "orthologue" parameter. This causes the portal to process the mouse genes as homologues instead of mouse genes.

You end up with 5 mouse genes. Here is the list I started with in humanmine:

```
ENSG00000069667
ENSG00000099250
ENSG00000100345
ENSG00000100644
ENSG00000102678
ENSG00000105974
ENSG00000106511
ENSG00000108691
ENSG00000110955
ENSG00000111145
ENSG00000112715
ENSG00000113196
ENSG00000116016
ENSG00000116678
ENSG00000119630
ENSG00000120899
ENSG00000121075
ENSG00000124882
ENSG00000125266
ENSG00000125968
ENSG00000134086
ENSG00000138792
ENSG00000140105
ENSG00000140545
ENSG00000143369
ENSG00000143878
ENSG00000143995
ENSG00000146904
ENSG00000150630
ENSG00000150782
ENSG00000151651
ENSG00000152234
ENSG00000153560
ENSG00000154188
ENSG00000160293
ENSG00000161714
ENSG00000163251
ENSG00000164107
ENSG00000164736
ENSG00000165197
ENSG00000169242
ENSG00000171951
ENSG00000172156
ENSG00000173511
ENSG00000177283
ENSG00000182580
ENSG00000185215
ENSG00000189030
ENSG00000196189
ENSG00000196411
ENSG00000203883
ENSG00000249915
```

I added an exception for mousemine so Rachel can finish the paper, and now it correctly finds all 56 genes.

**Comment by joshkh** [comment]
Flow:

1) **FriendlyMineLinkController.java** generates JSP variables:
```
${mines} = a list of know mines
${localMine} = the current mine
${identifiers} = identifiers from 
${organisms} = the organism from which the identifiers were resolved.
```
Example:
```
${mines} = mousemine, flymine, zfinmine
${localMine} = humanmine
${identifiers} = ENSG0001, ENSG0002, ENSG0003
${organisms} = H. sapien
```
* * *
2) **FriendlyMinesLinkDisplayer.jsp** generates javascript to run FriendlyMines.getLinks(...) on each remote mine.

For each mine in ${mines}, do the following:
```javascript
var mine = {name: '${mine.name}', url: '${mine.url}'};
var req = {origin: '${localMine.name}', domain: '${organisms}', identifiers: '${identifiers}'};
FriendlyMines.getLinks('#partner_mine_${mine.name}', mine, req);
```
Example:
```javascript
var mine = {name: 'mousemine', url: 'http://www.mousemine.org/mousemine'};
var req = {origin: 'humanmine', domain: "H. sapien", identifiers: ['ENSG0001', 'ENSG0002', 'ENSG0003']};
FriendlyMines.getLinks('#partner_mine_flymine', mine, req);
```
* * *
3) FriendlyMines.getLinks() calls **AjaxServices.getFriendlyMineLinks**:
```javascript
AjaxServices.getFriendlyMineLinks(mine.name, request.domain, request.identifiers, handlers);
```
Example
```javascript
AjaxServices.getFriendlyMineLinks("mousemine", "H. sapien", ['ENSG0001', 'ENSG0002', 'ENSG0003'], handlers);
```
* * *
4) AjaxServices.getFriendlyMineLinks does the following...
Get the info the remote mine.
Creates an empty collection of PartnerLinks.
Creates an ObjectRequest with the local organism and the list of local genes.
Creates a FriendlyMineManager. Not cached? Then...
Creates an InterMineLinkGenerator

(More to come)...

**Comment by julie-sullivan** [comment]
Fixed by f10b0b1d2c83672db53934e52a978fd6028f288f
